src/error-factory.js

```diff
@@ -1,11 +1,18 @@
 const util = require('util');
 
-const errorFactory = (name, message, baseType) => {
+const errorFactory = (name, parameters, baseType) => {
   if (baseType && baseType !== Error) {
     if ((typeof baseType) !== 'function' || (!(baseType.prototype instanceof Error))) {
       throw new Error('baseType prototype should be an instance of Error');
     }
   }
+  const [
+    message,
+    statusCode = (baseType ? baseType.statusCode : undefined),
+  ] = Array.isArray(parameters)
```

Comment:
Why array instead of an object? It might be more future proof to move baseType and parameters into an object.

```js
const errorFactory = (name, parameters, baseType)

// alongside with 

const errorFactory = ({ name, message, statusCode, baseType })
```

And have a check for function's arity